pkg/controller/cronjob/cronjob_controller_test.go

```diff
@@ -612,152 +614,158 @@ func TestSyncOne_Status(t *testing.T) {
-			if err != nil {
-				t.Errorf("%s: test setup error: failed to get job's ref: %v.", name, err)
+			if tc.ranPreviously {
+				sj.ObjectMeta.CreationTimestamp = metav1.Time{Time: justBeforeThePriorHour()}
+				sj.Status.LastScheduleTime = &metav1.Time{Time: justAfterThePriorHour()}
+			} else {
+				if tc.hasFinishedJob || tc.hasUnexpectedJob || tc.hasMissingJob {
+					t.Errorf("%s: test setup error: this case makes no sense", name)
+				}
+				sj.ObjectMeta.CreationTimestamp = metav1.Time{Time: justBeforeTheHour()}
 			}
-			sj.Status.Active = []v1.ObjectReference{*ref}
-			jobs = append(jobs, finishedJob)
-		}
-		if tc.hasUnexpectedJob {
-			jobs = append(jobs, unexpectedJob)
-		}
-		if tc.hasMissingJob {
-			ref, err := getRef(&missingJob)
-			if err != nil {
-				t.Errorf("%s: test setup error: failed to get job's ref: %v.", name, err)
+			jobs := []batchv1.Job{}
+			if tc.hasFinishedJob {
+				ref, err := getRef(&finishedJob)
+				if err != nil {
+					t.Errorf("%s: test setup error: failed to get job's ref: %v.", name, err)
+				}
+				sj.Status.Active = []v1.ObjectReference{*ref}
+				jobs = append(jobs, finishedJob)
+			}
+			if tc.hasUnexpectedJob {
+				jobs = append(jobs, unexpectedJob)
+			}
+			if tc.hasMissingJob {
+				ref, err := getRef(&missingJob)
+				if err != nil {
+					t.Errorf("%s: test setup error: failed to get job's ref: %v.", name, err)
+				}
+				sj.Status.Active = append(sj.Status.Active, *ref)
+			}
+			if tc.beingDeleted {
+				timestamp := metav1.NewTime(tc.now)
+				sj.DeletionTimestamp = &timestamp
 			}
-			sj.Status.Active = append(sj.Status.Active, *ref)
-		}
-		if tc.beingDeleted {
-			timestamp := metav1.NewTime(tc.now)
-			sj.DeletionTimestamp = &timestamp
-		}
 
-		jc := &fakeJobControl{}
-		sjc := &fakeSJControl{}
-		recorder := record.NewFakeRecorder(10)
+			jc := &fakeJobControl{}
+			sjc := &fakeSJControl{}
+			recorder := record.NewFakeRecorder(10)
 
-		// Run the code
-		syncOne(&sj, jobs, tc.now, jc, sjc, recorder)
+			// Run the code
+			syncOne(&sj, jobs, tc.now, jc, sjc, recorder)
 
-		// Status update happens once when ranging through job list, and another one if create jobs.
-		expectUpdates := 1
-		// Events happens when there's unexpected / finished jobs, and upon job creation / deletion.
-		expectedEvents := 0
-		if tc.expectCreate {
-			expectUpdates++
-			expectedEvents++
-		}
-		if tc.expectDelete {
-			expectedEvents++
-		}
-		if tc.hasFinishedJob {
-			expectedEvents++
-		}
-		if tc.hasUnexpectedJob {
-			expectedEvents++
-		}
-		if tc.hasMissingJob {
-			expectedEvents++
-		}
+			// Status update happens once when ranging through job list, and another one if create jobs.
+			expectUpdates := 1
+			// Events happens when there's unexpected / finished jobs, and upon job creation / deletion.
+			expectedEvents := 0
+			if tc.expectCreate {
+				expectUpdates++
+				expectedEvents++
+			}
+			if tc.expectDelete {
+				expectedEvents++
+			}
+			if tc.expectMissedStart {
```

Comment:
This "count the events" part of the test is starting to get a little goofy, but I guess it's not worth ripping the bandage off now.